pkgs/development/compilers/rust/rustc.nix

```diff
@@ -59,28 +52,58 @@ stdenv.mkDerivation {
 
   # We need rust to build rust. If we don't provide it, configure will try to download it.
   # Reference: https://github.com/rust-lang/rust/blob/master/src/bootstrap/configure.py
-  configureFlags = configureFlags
-                ++ [ "--enable-local-rust" "--local-rust-root=${rustPlatform.rust.rustc}" "--enable-rpath"
-                     "--enable-vendor"
-                     "--jemalloc-root=${prefixedJemalloc}/lib"
-                     "--default-linker=${targetPackages.stdenv.cc}/bin/cc" ]
-                ++ optional (!withBundledLLVM) [ "--enable-llvm-link-shared" "--llvm-root=${llvmShared}" ]
-                ++ optional (targets != []) "--target=${target}";
+  configureFlags = let
+    setHost = "--set=target.${stdenv.hostPlatform.config}";
+    hostCC = "${buildPackages.stdenv.cc}/bin/cc";
+    setTarget = "--set=target.${stdenv.targetPlatform.config}";
+    targetCC = "${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}cc";
```

Comment:
"build" "host" and "target" are confused here again. `rustc`, like all tools, should be taken from the previous bootstrapping stage.

Reply:
I can elaborate further e.g. on IRC, but you can look at the android and iOS cross prebuilt packages for an example of how this stage stuff works unencumbered by other details.